Kernel/LookaheadLiteralSelector.cpp

```diff
@@ -46,7 +47,7 @@ using namespace Saturation;
  */
 struct LookaheadLiteralSelector::GenIteratorIterator
 {
-  DECL_ELEMENT_TYPE(VirtualIterator<void>);
+  DECL_ELEMENT_TYPE(VirtualIterator<std::tuple<>>);
```

Comment:
I think it's some form of an abstraction where the lookahead selection iterates through the indexes like if it were performing inferences on one of the yet to be selected literals and is only interested in COUNTING how many hits it would get. (While the iterators in their original role actually generate various things - substitutions, clauses. And there are just being forgotten here.

Reply:
I'm still not clear why it's now empty-tuple rather than void, but makes sense to me and we have bigger fish to fry.

Reply:
I think it's simply because you can't write `return void` or `return something` that would type as `void`. Or something like that ;)